sapling/src/SidebarProvider.ts

```diff
@@ -72,14 +72,23 @@ export class SidebarProvider implements vscode.WebviewViewProvider {
       switch (data.type) {
         // Case when the user selects a file to begin a tree
         case "onFile": {
-          // Edge case if the user sends in nothing
-          if (!data.value) {
-            return;
-          }
-          // Run an instance of the parser
-          this.parser = new SaplingParser(data.value);
-          this.parser.parse();
-          this.updateView();
+          // open vscode dialog selector
+          vscode.window.showOpenDialog({canSelectMany: false, canSelectFolders: false})
+            .then( uri => {
+            
+            // Edge case if selector doesn't work
+            if (!(uri && uri[0])) {
```

Comment:
According to docs, showOpenDialog either resolves to the resources or returns undefined. Why do we need to check uri[0] in addition to uri, instead of just if uri is truthy?

Reply:
the resources resolve to either an array or undefined